ts/sagas/profile.ts

```diff
@@ -4,98 +4,98 @@
 import { call, Effect, put, select, takeLatest } from "redux-saga/effects";
 
 import {
-  ApiFetchResult,
-  ApiProfile,
-  fetchProfile,
-  isApiFetchFailure,
-  postProfile
-} from "../api";
+  BackendClient,
+  BasicResponseTypeWith401,
+  ProfileWithOrWithoutEmail
+} from "../api/backend";
+import { apiUrlPrefix } from "../config";
 import {
-  PROFILE_LOAD_FAILURE,
   PROFILE_LOAD_REQUEST,
-  PROFILE_LOAD_SUCCESS,
-  PROFILE_UPDATE_FAILURE,
-  PROFILE_UPDATE_REQUEST,
-  PROFILE_UPDATE_SUCCESS
+  PROFILE_UPSERT_REQUEST
 } from "../store/actions/constants";
-import { ProfileUpdateRequest } from "../store/actions/profile";
+import {
+  profileLoadFailure,
+  profileLoadSuccess,
+  profileUpsertFailure,
+  ProfileUpsertRequest,
+  profileUpsertSuccess
+} from "../store/actions/profile";
 import { sessionTokenSelector } from "../store/reducers/authentication";
 import { SessionToken } from "../types/SessionToken";
+import { callApiWith401ResponseStatusHandler } from "./api";
 
 // A saga to load the Profile.
 function* loadProfile(): Iterator<Effect> {
-  try {
-    // Get the token from the state
-    const sessionToken: SessionToken | undefined = yield select(
-      sessionTokenSelector
-    );
-    if (sessionToken === undefined) {
-      throw new Error("session token is not defined");
-    }
+  const sessionToken: SessionToken | undefined = yield select(
+    sessionTokenSelector
+  );
 
-    // Fetch the profile from the proxy
-    const response: ApiFetchResult<ApiProfile> = yield call(
-      fetchProfile,
-      sessionToken
+  if (sessionToken) {
+    const backendClient = BackendClient(apiUrlPrefix, sessionToken);
+
+    const response:
+      | BasicResponseTypeWith401<ProfileWithOrWithoutEmail>
+      | undefined = yield call(
+      callApiWith401ResponseStatusHandler,
+      backendClient.getProfile,
+      {}
     );
 
-    if (isApiFetchFailure(response)) {
-      // If the api response is an error then dispatch the PROFILE_LOAD_FAILURE action.
-      yield put({
-        type: PROFILE_LOAD_FAILURE,
-        payload: response.error
-      });
+    if (!response || response.status !== 200) {
+      // We got a error, send a SESSION_LOAD_FAILURE action
+      const error: Error = response ? response.value : Error();
```

Comment:
do we have an error message for `Error()` ?

Reply:
Same as above